### PR TITLE
added order and orderItem

### PR DESCRIPTION
**Pull request by vladshender**

### CODE REVIEW

File: src/main/java/com/example/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public OrderResponseDto exchangeStatus(
+            @PathVariable Long id,
+            @RequestBody @Valid OrderUpdateStatusRequestDto requestDto
+    ) {
+        return orderService.updateStatusByOrder(id, requestDto);
+    }
+
+    @PreAuthorize("hasAuthority('ROLE_USER')")
+    @Operation(summary = "Get all items for order by order id")
+    @GetMapping("/{id}/items")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/controller/OrderController.java:60]:
```suggestion
    @GetMapping("/{orderId}/items")
```

File: src/main/java/com/example/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public List<OrderItemResponseDto> getAllItemsByOrderId(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/controller/OrderController.java:61]:
```suggestion
    public List<OrderItemResponseDto> getAllItemsByOrderId(@PathVariable Long orderId) {
```

File: src/main/java/com/example/service/order/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderResponseDto> getOrderByUser(User user) {
+        List<Order> orders = orderRepository.findAllByUser(user);
+        if (orders.isEmpty()) {
+            throw new EntityNotFoundException(
+                    "Can't find order list for user by id: " + user.getId());
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/order/OrderServiceImpl.java:67]:
```suggestion
```

File: src/main/java/com/example/service/order/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            OrderUpdateStatusRequestDto responseDto) {
+        Order orderByUser = findById(id);
+        orderMapper.updateOrderFromDto(responseDto, orderByUser);
+        return orderMapper.toDto(orderRepository.save(orderByUser));
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemResponseDto> getOrderItemsByOrderId(Long id) {
+        Order order = findById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/order/OrderServiceImpl.java:83]:
```suggestion
        Order order = orderRepository.findByIdAndUserId(orderId, userId).orElseThrow(()...
        return orderItemMapper.toOrderItemDtoList(order.getOrderItems());
```

File: src/main/java/com/example/service/order/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        return orderItemMapper.toDtoList(orderItemsRepository.findAllByOrder(order));
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemFromOrderById(Long orderId, Long orderItemsId) {
+        Order order = findById(orderId);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/service/order/OrderServiceImpl.java:89]:
```suggestion
        OrderItem item = itemRepository.findByIdAndOrderIdAndUserId(orderItemId, orderId, userId)....
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Comment by vladshender** [comment]
according to your last comment, order item does not have a user field, so searching the database by user id is impossible

**Pull request merged**